lib/route_translator/extensions/route_set.rb

```diff
@@ -5,12 +5,42 @@
 module ActionDispatch
   module Routing
     class RouteSet
+      module VerificationLambdas
+        module_function
+
+        def for_locale(locale)
+          locale = sanitize_locale(locale)
+
+          lambdas[locale] ||= ->(req) { locale == RouteTranslator::Host.locale_from_host(req.host).to_s }
+        end
+
+        def lambdas
+          @lambdas ||= {}
+        end
+
+        def sanitize_locale(locale)
+          locale.to_s.gsub('native_', '')
+        end
+      end
+
       def add_localized_route(mapping, path_ast, name, anchor, scope, path, controller, default_action, to, via, formatted, options_constraints, options)
```

Comment:
```
lib/route_translator/extensions/route_set.rb:26:7: C: Metrics/AbcSize: Assignment Branch Condition size for add_localized_route is too high. [26.89/23.15]
      def add_localized_route(mapping, path_ast, name, anchor, scope, path, controller, default_action, to, via, formatted, options_constraints, options)
```

This is going to add complexity to a method which is already very complex. Any chance to split this?